store/tikv/lock_resolver.go

```diff
@@ -421,10 +421,12 @@ func (lr *LockResolver) resolveLocks(bo *Backoffer, callerStartTS uint64, locks
 				}
 			} else {
 				if status.action != kvrpcpb.Action_MinCommitTSPushed {
-					pushFail = true
-					return nil
+					if _, exists := pushed[l.TxnID]; !exists {
```

Comment:
It's a bug in TiKV. Let me fix it. However, we can avoid pushing locks of the same transaction with the idea here.

Reply:
Do we have an issue for it?

Reply:
No issue in TiKV repo. https://github.com/pingcap/tidb/issues/21658 is enough and this bug only exists in v5.0.

Reply:
https://github.com/tikv/tikv/pull/9319

Reply:
Is it change necessary if tikv/tikv#9319 is merged?

Reply:
I think it seems no necessary - - but maybe we can try and test avoid pushing locks in future